game-main/src/main.rs

```diff
@@ -38,7 +38,8 @@ fn main() -> amethyst::Result<()> {
         .with_bundle(
             InputBundle::<String, String>::new()
                 .with_bindings_from_file(format!("{}/input.ron", root))?,
-        )?.with(player::Movement, "player-movement", &[])
+        )?
+        .with(player::Movement, "player-movement", &[])
```

Comment:
Not this line specifically but note that there are empty arrays in these declarations, these are dependencies. If you split the grouper and movement into two different systems you can specify that the ally::Movement system depends on the ally::Grouper system, so that it will be run after it, better separation of mutability helps with amethyst's ability to parallelize the systems.